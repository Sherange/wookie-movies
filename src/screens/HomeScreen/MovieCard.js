import React from 'react';
import {Image, StyleSheet, Text, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import {primaryTextColor} from '../../constants/theme';

const MovieCard = props => {
  const {poster, title, length, classification} = props.item;
  return (
    <TouchableOpacity
      style={styles.movieCardContainer}
      onPress={() => props.navigateDetailScreen(props.item)}>
      <Image source={{uri: poster}} style={{width: 150, height: 250}}></Image>
      <Text style={styles.titleStyle} numberOfLines={1}>
        {title}
      </Text>
      <Text style={styles.subTitleStyle} numberOfLines={1}>
        {length} | {classification}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  movieCardContainer: {
    width: 150,
    marginRight: 10,
    marginVertical: 8,
  },
  titleStyle: {
    marginVertical: 4,
    color: primaryTextColor,
    fontSize: 16,
    fontWeight: '600',
  },
  subTitleStyle: {
    marginVertical: 2,
    color: primaryTextColor,
    opacity: 0.6,
    fontSize: 14,
  },
});

MovieCard.propTypes = {
  item: PropTypes.shape({
    poster: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    length: PropTypes.string.isRequired,
    classification: PropTypes.string.isRequired,
  }),
  navigateDetailScreen: PropTypes.func.isRequired,
};

export default MovieCard;
