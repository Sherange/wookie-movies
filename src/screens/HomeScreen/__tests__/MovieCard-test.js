import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from '../MovieCard';

const testData = {
  classification: '13+',
  length: '2h 32min',
  poster:
    'https://wookie.codesubmit.io/static/posters/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg',
  title: 'The Dark Knight',
};

it('renders correctly', () => {
  renderer.create(
    <MovieCard
      image={testData.poster}
      title={testData.title}
      classification={testData.classification}
      length={testData.length}
      navigateDetailScreen={() => {}}
    />,
  );
});

test('MovieCard snapshot', () => {
  const snapshot = renderer
    .create(
      <MovieCard
        image={testData.poster}
        title={testData.title}
        classification={testData.classification}
        length={testData.length}
        navigateDetailScreen={() => {}}
      />,
    )
    .toJSON();
  expect(snapshot).toMatchSnapshot();
});
