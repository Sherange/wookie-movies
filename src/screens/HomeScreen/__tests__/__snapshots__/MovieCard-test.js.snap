// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 10,
      "marginVertical": 8,
      "opacity": 1,
      "width": 150,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://wookie.codesubmit.io/static/posters/d6822b7b-48bb-4b78-ad5e-9ba04c517ec8.jpg",
      }
    }
    style={
      Object {
        "height": 250,
        "width": 150,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
        "fontWeight": "600",
        "marginVertical": 4,
      }
    }
  >
    The Dark Knight
  </Text>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 14,
        "marginVertical": 2,
        "opacity": 0.6,
      }
    }
  >
    2h 32min
     | 
    13+
  </Text>
</View>
`;
